pkg/client/cache/delta_fifo.go

```diff
@@ -522,6 +522,23 @@ func (f *DeltaFIFO) syncKey(key string) error {
 		return nil
 	}
 
+	// If we are doing Resync() and there is already an event queued for that object,
```

Comment:
Ideally, I think that we should introduce a new type of event "Replace" (next to Added, Updated, Deleted and Sync), but maybe not in 1.4 timeframe?

WDYT?